File: 2900-longest-unequal-adjacent-groups-subsequence-i/2900-longest-unequal-adjacent-groups-subsequence-i.cpp
class Solution {
public:
    vector<string> getLongestSubsequence(vector<string>& words, vector<int>& groups) {
        vector<string> subsequence;
        if (words.empty()) return subsequence;

        subsequence.push_back(words[0]);
        for (int i=1; i<words.size(); ++i) {
            if (groups[i] != groups[i-1]) {
                subsequence.push_back(words[i]);
            }
        }
        return subsequence;
    }
};

File: 2900-longest-unequal-adjacent-groups-subsequence-i/2900-longest-unequal-adjacent-groups-subsequence-i.rs
impl Solution {
    pub fn get_longest_subsequence(words: Vec<String>, groups: Vec<i32>) -> Vec<String> {
        let mut subsequence = Vec::new();
        subsequence.push(words[0].clone());

        for i in 1..words.len() {
            if groups[i] != groups[i-1] {
                subsequence.push(words[i].clone());
            }
        }
        subsequence
    }
}
